solara/components/file_drop.py

```diff
@@ -94,3 +94,59 @@ def handle_file(cancel: threading.Event):
         raise result.error
 
     return file_drop
+
+
+@solara.component
+def FileDropMultiple(
+    label="Drop files here",
+    on_total_progress: Optional[Callable[[float], None]] = None,
+    on_file: Optional[Callable[[List[FileInfo]], None]] = None,
+    lazy: bool = True,
+) -> ipyvuetify.extra.FileInput:
+    """Region a user can drop multiple files into for file uploading.
+
+    Almost identical to `FileDrop` except that `on_file` is called
+    with a list of `FileInfo` objects.
+
+    ## Arguments
+     * `on_total_progress`: Will be called with the progress in % of the file(s) upload.
+     * `on_file`: Will be called with a `List[FileInfo]`.
+        Each `FileInfo` contains the file `.name`, `.length`, `.file_obj` object, and `.data` attributes.
+     * `lazy`: Whether to load the file contents into memory or not.
+
+    """
+    file_info, set_file_info = solara.use_state(None)
+    wired_files, set_wired_files = solara.use_state(cast(Optional[typing.List[FileInfo]], None))
+
+    file_drop = FileDropZone.element(label=label, on_total_progress=on_total_progress, on_file_info=set_file_info, multiple=True)  # type: ignore
+
+    def wire_files():
+        if not file_info:
+            return
+
+        real = cast(FileDropZone, solara.get_widget(file_drop))
+
+        # workaround for @observe being cleared
+        real.version += 1
+        real.reset_stats()
+
+        set_wired_files(cast(typing.List[FileInfo], real.get_files()))
+
+    solara.use_side_effect(wire_files, [file_info])
+
+    def handle_file(cancel: threading.Event):
+        if not wired_files:
+            return
+        if on_file:
+            for i in range(len(wired_files)):
+                if not lazy:
+                    wired_files[i]["data"] = wired_files[i]["file_obj"].read()
+                else:
+                    wired_files[i]["data"] = None
+            on_file(wired_files)
+
+    result: solara.Result = hooks.use_thread(handle_file, [wired_files])
+    if result.error:
+        raise result.error
+
+    return file_drop
```

Comment:
I don't think we want this file duplication. I'm sorry if I wasn't clear in my previous message.
Basically what I suggested is that your previous implementation could be used by these two implementations. So lets say we call your previous component `_FileDrop`, we simply call `return _FileDrop(...., multiple=True)` from this component.